doc/api/async_hooks.md

```diff
@@ -712,6 +712,88 @@ never be called.
 * Returns: {number} The same `triggerAsyncId` that is passed to the
 `AsyncResource` constructor.
 
+## Class: AsyncContext
+<!-- YAML
+added: REPLACEME
+-->
+
+This class is used to create asynchronous state between callbacks and promise
+chains. It allows you to keep variables throughout the lifetime of a web request
+or any other asynchronous duration. It is similar to what ThreadLocal storage is
+ in other languages.
+
+### Example
```

Comment:
I don't think you need a separate header for the example - just start a new paragraph.